test/junit/scala/collection/SeqTest.scala

```diff
@@ -90,4 +90,11 @@ class SeqTest {
     assert(s.lengthIs > 2)
     assert(s.lengthIs != 2)
   }
+
+  @Test
+  def testEndsWith(): Unit = {
+    assertFalse((0 until 0).endsWith(List(-4, -3, -2, -1)))
+    assertTrue((-8 until -1).endsWith((-8 until -1).takeRight(1)))
+    assertTrue((0 until 0).endsWith(0 until 0))
+  }
```

Comment:
I would move this test into `test/junit/scala/collection/immutable/RangeTest.scala` (if it doesn't exist, you can create it), since the bug and tests are specifically to do with `Range`. It might also be good to add some tests specifically for `RangeIterator#drop` (e.g. `assertEquals(Nil, (0 until 0).iterator.drop(-1).toList)`.

Reply:
I've moved the tests over to Range and added additional ones for testing RangeIterator#drop.